client/finality-grandpa/src/communication/mod.rs

```diff
@@ -628,10 +629,11 @@ pub struct SetId(pub SetIdNumber);
 pub(crate) struct OutgoingMessages<Block: BlockT> {
 	round: RoundNumber,
 	set_id: SetIdNumber,
-	locals: Option<(AuthorityPair, AuthorityId)>,
+	local_id: Option<AuthorityId>,
 	sender: mpsc::Sender<SignedMessage<Block>>,
 	network: Arc<Mutex<GossipEngine<Block>>>,
 	has_voted: HasVoted<Block>,
+	keystore: Option<BareCryptoStorePtr>,
```

Comment:
Should we instead have `Option<(AuthorityId, BareCryptoStorePtr)>`? If we have defined a `local_id` then we must have a keystore as well, and we can turn this into a compile error.

Reply:
I went ahead and implemented this to find out that i am storing the keystore in `Config` as `Option` there where `Some(keystore)` would be if the node `is_authority`. So i would need to `match` anyway here: https://github.com/paritytech/substrate/pull/6178/files#diff-b2091839f6f331f9cb810c8c37ccb44cR720